dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -324,3 +324,18 @@ def unalikeability(self) -> float | None:
             ) * self._categories[category]
         unalike: float = unalike_sum / (self.sample_size**2 - self.sample_size)
         return unalike
+
+    def to_dict(self) -> dict:
+        """
+        Create serializable dict of this object.
+
+        :return: dict
+        """
+        categorical_dict = {
+            "type": self.type,
+            "_categories": self._categories,
+            "__calculations": self.__calculations,
+            "_top_k_categories": self._top_k_categories,
+        }
+        categorical_dict.update(super().to_dict())
```

Comment:
As opposed to updating the dict, there could be overlap issues.
What if we do some mechanic to store it in a key referencing the base class?

Reply:
This would allow us to also know where to route on load.